### PR TITLE
Fix compiler-rt-sanitizers to be able to build on arm targets

### PR DESCRIPTION
**Pull request by fwuehr95**
Signed off: Florian Wühr [fwuehr@rational-online.com](mailto:fwuehr@rational-online.com)
Signed off: Alexander Thoma [athoma@rational-online.com](mailto:athoma@rational-online.com)
---
### Contributor checklist

- [x] Changes have been tested
- [x] `Signed-off-by` is present
- [x] The PR complies with the [Open Embedded Commit Patch Message Guidelines](http://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines)

### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)

### CODE REVIEW

**Comment by fwuehr95** [comment]
We noticed that for arm32 or arm64 based targets it is not possible to build compiler-rt-sanitizers without adding "-fPIC" to CXXFLAGS. For x86-64 based targets it looks like it doesn´t really matter. Introducing "-fPIC" leads to package_qa errors with relocations in .text sections, which we have disable by adding "textrel" to INSANE_SKIP. This looks to be a known issue, other layer maintainers such as on meta-virtualization and openembedded-core used the same way here.

Additionally we changed the default value of RUNTIME variable, which has been set hardcoded to "llvm" prior to our change. According to meta-clang layer documentation RUNTIME should be set to "gnu" as default. Assigning "gnu" only with "?=" instead of "=" assignment allows to set RUNTIME variable to the desired value in a central conf file such as distro or machine configuration and to be used as a global variable. 

This helped us fixing the issue that we are not able of compiling libcxx for our arm32 and arm64 targets and to use GNU libstdc++ instead.

File: recipes-devtools/clang/compiler-rt-sanitizers_git.bb [hunk 1]
```diff
@@ -15,12 +15,18 @@ LIC_FILES_CHKSUM = "file://compiler-rt/LICENSE.TXT;md5=d846d1d65baf322d4c485d6ee
 
 TUNE_CCARGS_remove = "-no-integrated-as"
 
-RUNTIME = "llvm"
+#compiler-rt-sanitizers will not compile for arm32 or arm64 based platforms without adding -fPIC
+CXXFLAGS:append = " -fPIC"
+
+#set RUNTIME variable according to documention to gnu as default value
+RUNTIME ?= "gnu"
```
Comment (kraj) [inline recipes-devtools/clang/compiler-rt-sanitizers_git.bb:22]: gnu is default runtime, so we don't need to repeat it here.

**Review by kraj — CHANGES_REQUESTED** [review]

File: recipes-devtools/clang/compiler-rt-sanitizers_git.bb (hunk 1 above)
Comment (fwuehr95) [reply recipes-devtools/clang/compiler-rt-sanitizers_git.bb:22]: Fine. Why has it been set to "llvm" previous to my changes though ?

Comment (kraj) [reply recipes-devtools/clang/compiler-rt-sanitizers_git.bb:22]: this recipe is derived from compiler-rt which obviously is part of llvm runtime but sanitizers could be decoupled from runtime, so I am ok removing it.

**Pull request closed without merging**